errors.go

```diff
@@ -109,6 +124,40 @@ func sprintf(format string, args ...interface{}) string {
 	return fmt.Sprintf(format, args...)
 }
 
+// NewBatch creates a new BatchError.
+func NewBatch() BatchError {
+	return &batcherrtype{}
+}
+
+//Append appends the error
+func (b *batcherrtype) Append(err error) {
+	b.errors = append(b.errors, err)
+}
+
+//Errors returns the errors
+func (b *batcherrtype) Errors() []error {
+	return b.errors
+}
+
+//IsEmpty returns if there is any error
+func (b *batcherrtype) IsEmpty() bool {
+	return len(b.errors) == 0
+}
```

Comment:
Perhaps we can do as `awserr` does and only accept an array of errors (or `...`). Then we wouldn't need to check `IsEmpty()` as the instance itself would mean there is an error. Otherwise it would be `nil`.